service/src/PlatformIO.cpp

```diff
@@ -210,26 +227,49 @@ namespace geopm
         if (sig_tup_it != m_existing_signal.end()) {
             result = sig_tup_it->second;
         }
+        std::string err_msg;
+        bool read_signal_ok = false;
```

Comment:
Move this declaration closer to use inside of for and if contexts, i.e. after line 235